src/vector/layer.rs

```diff
@@ -620,7 +619,8 @@ impl Dataset {
     ///
     /// Applies to vector datasets, and fetches by the given
     /// _0-based_ index.
-    pub fn into_layer(self, idx: isize) -> Result<OwnedLayer> {
+    pub fn into_layer(self, idx: usize) -> Result<OwnedLayer> {
+        let idx = i32::try_from(idx)?;
```

Comment:
I didn't check, but does `c_int::try_from` work? It might be more portable, and we cast to `c_int` afterwards anyway.

Ditto everywhere else.

Reply:
`libc::c_int::try_from` does work! Good idea.